src/coreclr/jit/morph.cpp

```diff
@@ -5120,6 +5512,12 @@ void Compiler::fgMoveOpsLeft(GenTree* tree)
             noway_assert(op1->gtType == TYP_I_IMPL && ad1->gtType == TYP_I_IMPL);
             new_op1->gtType = TYP_I_IMPL;
         }
+#ifdef TARGET_LOONGARCH64
+        else if ((op1->TypeGet() == TYP_LONG) && (ad2->TypeGet() == TYP_INT))
+        {
+            new_op1->gtType = TYP_LONG;
+        }
+#endif
```

Comment:
Please add a comment (here and in code) why this is required for LA. There is no other arch-specific code in this function.

Reply:
```
+            // For LoongArch64's instructions operation of the 64bits and 32bits using the whole
+            // 64bits-width register which is unlike the AMD64 and ARM64.
+            // And the INT type instruction will be signed-extend by default.
+            // e.g. 'ld_w $r4, $5, 4' and `addi_w  $r4,$r5,-1` the result of INT
+            // will be signed-extend by default.
```